tests/unit/src/test/resources/semanticTokens/example/ForComprehensions.scala

```diff
@@ -1,31 +1,31 @@
 <<package>>/*keyword*/ <<example>>/*namespace*/
 
-<<class>>/*keyword*/ <<ForComprehensions>>/*class*/ {
+<<class>>/*keyword*/ <<ForComprehensions>>/*class,definition*/ {
   <<for>>/*keyword*/ {
-    <<a>>/*parameter*/ <<<->>/*operator*/ <<List>>/*variable,readonly*/(<<1>>/*number*/)
-    <<b>>/*parameter*/ <<<->>/*operator*/ <<List>>/*variable,readonly*/(<<a>>/*parameter*/)
+    <<a>>/*parameter,declaration*/ <<<->>/*operator*/ <<List>>/*variable,readonly*/(<<1>>/*number*/)
+    <<b>>/*parameter,declaration*/ <<<->>/*operator*/ <<List>>/*variable,readonly*/(<<a>>/*parameter*/)
     <<if>>/*keyword*/ (
       <<a>>/*parameter*/,
       <<b>>/*parameter*/,
     ) <<==>>/*method*/ (<<1>>/*number*/, <<2>>/*number*/)
     (
-      <<c>>/*variable,readonly*/,
-      <<d>>/*variable,readonly*/,
+      <<c>>/*variable,definition,readonly*/,
+      <<d>>/*variable,definition,readonly*/,
```

Comment:
This feels a bit counter-intuitive to me. Assigning a value in a for comprehension is a `declaration` but if you do unapply it's a `definition`... How will this work for pattern matching?
```
some match {
    case s/* declaration ? */ @ Some(number/* definition ? */) =>
    case n/* definition  or declaration ? */ =>
  }
```

Reply:
I agree it can be counter-intuitive sometimes, but in pattern matching it looks good. In example above, `s`, `number` and `n` become `definition`. It will become `declaration` in e.g. `x.map(<<abc>>/*declaration*/ => ...)`

Reply:
Ahh... I see how that makes sense. Thanks.

Reply:
It's the same as declaration of anonymous function argument, we can probably check with sth like `(sym.is(Flags.Parameter) && sym.owner.isAnonymousFunction)`, but I think current approach is better

Reply:
No, I see why it's this way. If anything I'll do it the other way around:
```
val m: Int => Int = {
    case v /* declaration*/ => v
  }
```
So if you have a lambda and you pattern match without binding the arg first the matched parts **could** all be declarations (since truly you don't have a value yet). But I don't see that being exactly less confusing and I'm pretty sure it's not worth it.

Reply:
I think @kasiaMarek is right, looking into some more formal definitions:
```
Declaration means that variable is only declared and memory is allocated, but no value is set.
```
so in pattern matching we don't really have the value set yet.

Not sure how to deal with classes/object, but most likely I should remove both from there to avoid misunderstandings.

Reply:
Ok, I changed the behaviour to be a bit less controversial and I added some comments. Let me know what you think!